Add order model

waspp01 opened the pull request.

waspp01 committed:
#9 - Order and OrderItem addition:
- Added Order entity, Service, Repository, DTOs , Controller, Mapper
- Added OrderItem entity, DTOs, Repository, Mapper
- Added deleteAllByShoppingCartId to CartItemRepository
- Added cleanShoppingCart method to ShoppingCartService and its Impl
- Added required Liquibase files

Rommelua commented on src/main/java/com/example/onlinebookstore/controller/ShoppingCartController.java:
```diff
@@ -0,0 +1,73 @@
+
+@Tag(name = "Shopping cart management", description = "Endpoints for managing shopping cart")
+@RestController
+@RequestMapping("/cart")
```
```suggestion
@RequestMapping("/carts")
```

Rommelua commented on src/main/java/com/example/onlinebookstore/controller/ShoppingCartController.java:
```diff
@@ -0,0 +1,73 @@
+@RestController
+@RequestMapping("/cart")
+@RequiredArgsConstructor
+public class ShoppingCartController {
```
ShoppingCartController is not a part of this task

sokimaaa replied:
not fixed

koval-kn replied:
not fixed

waspp01 committed:
#10 - Order and OrderItem addition:
Changed "/carts" endpoint naming

waspp01 committed:
Revert "#10 - Order and OrderItem addition:"

This reverts commit 2e01af97b1e05778732901635bd2297fc8a62df1.

waspp01 committed:
Merge branch 'main' into add-order-model

koval-kn commented on src/main/java/com/example/onlinebookstore/model/Order.java:
```diff
@@ -0,0 +1,63 @@
+@Getter
+@Setter
+@Table(name = "orders")
+public class Order {
```
soft-delete approach was not implemented
![image](https://github.com/waspp01/online-book-store-app/assets/73041091/91486adf-1331-4345-aeb5-49453e89b5bb)

koval-kn commented on src/main/java/com/example/onlinebookstore/model/OrderItem.java:
```diff
@@ -0,0 +1,41 @@
+@Getter
+@Setter
+@Table(name = "order_items")
+public class OrderItem {
```
soft-delete approach was not implemented
![image](https://github.com/waspp01/online-book-store-app/assets/73041091/91486adf-1331-4345-aeb5-49453e89b5bb)

koval-kn commented on src/main/java/com/example/onlinebookstore/controller/OrderController.java:
```diff
@@ -0,0 +1,87 @@
+            description = "Get all info about the orders")
+    @ApiResponse(responseCode = "200", description = "Order information",
+            content = {@Content(mediaType = "application/json")})
+    public List<OrderDto> getAll(Authentication authentication) {
```
pagination was not implemented
![image](https://github.com/waspp01/online-book-store-app/assets/73041091/91486adf-1331-4345-aeb5-49453e89b5bb)

koval-kn commented on src/main/java/com/example/onlinebookstore/service/OrderService.java:
```diff
@@ -0,0 +1,20 @@
+public interface OrderService {
+    OrderDto createOrder(User user, CreateOrderRequestDto requestDto);
+
+    List<OrderDto> getAll(Long userId);
```
pagination was not implemented
![image](https://github.com/waspp01/online-book-store-app/assets/73041091/91486adf-1331-4345-aeb5-49453e89b5bb)

koval-kn commented on src/main/java/com/example/onlinebookstore/model/OrderItem.java:
```diff
@@ -0,0 +1,41 @@
+    @Column(nullable = false)
+    private BigDecimal price;
+
+
```
redundant blank lines
```suggestion
```

koval-kn commented on src/main/java/com/example/onlinebookstore/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,108 @@
+    }
+
+    @Override
+    public List<OrderDto> getAll(Long userId) {
```
pagination was not implemented
![image](https://github.com/waspp01/online-book-store-app/assets/73041091/91486adf-1331-4345-aeb5-49453e89b5bb)

waspp01 committed:
#11 - Order and OrderItem addition:
Implemented pagination for OrderController, OrderRepository, OrderService
Implemented soft delete for Order and OrderItem
Added respective changes to the liquibase

Rommelua commented on src/main/java/com/example/onlinebookstore/service/OrderItemService.java:
```diff
@@ -0,0 +1,4 @@
+package com.example.onlinebookstore.service;
+
+public interface OrderItemService {
```
What is it for?

Rommelua commented on src/main/java/com/example/onlinebookstore/service/impl/OrderItemServiceImpl.java:
```diff
@@ -0,0 +1,4 @@
+package com.example.onlinebookstore.service.impl;
+
+public class OrderItemServiceImpl {
```
???

waspp01 committed:
#12 - OrderItemService and OrderItemServiceImpl deleted
